craft_application/services/project.py

```diff
@@ -70,7 +70,7 @@ def configure(self, *, platform: str | None, build_for: str | None) -> None:
         """
         self._build_on = craft_platforms.DebianArchitecture.from_host()
         if self.is_configured:
-            raise RuntimeError("Project is already configured.")
+            raise errors.ProjectGenerationError("Project is already configured.")
```

Comment:
We should keep this as a RuntimeError (or similar) as configuring the project service twice is a bug.

Reply:
Rolled back this change.